// @ali/route-map-http/lib/index.js
'use strict';

const apiParser = require('./api-parser');
const CDN_Prefix = '//global.alicdn.com/';
const CDN_Suffix = '';

/**
 * @description 路由映射表
 * @param apiConfig {Object} 接口配置
 * @param apiConfig.api {String} 接口API名称（必须，url接口为url地址，mtop接口为mtop api名称）
 * @param apiConfig.data {Object} 接口参数（非必须）
 * @param apiConfig.appId {Number} 接口所属应用ID（非必须）
 * @param apiConfig.dirName {String} 接口目录（非必须，默认default）
 * @param apiConfig.requiredParams {Array} 指定接口必须入参（非必须，若未指定，默认认为所有入参为必须入参）
 */


function HttpRouteMap(apiConfig) {
  apiConfig.type = apiConfig.type || 'http';
  this.apiConfig = apiConfig;
  this.rules = apiParser.parse(apiConfig);
}

HttpRouteMap.prototype = {

  getKey: function() {
    return this.rules && this.rules.key ? ('TAI_LSKEY_' + this.apiConfig.type.toUpperCase() + '_' + this.rules.key) : null;
  },

  getCDNAddress: function() {
    return this.rules ? (CDN_Prefix + this.rules.path + (this.rules.params ? ('/' + this.rules.params) : '') + CDN_Suffix) : null;
  },

  getJSONPCallbak: function() {
    return this.rules.key || null;
  }
};

module.exports = HttpRouteMap;
