// @ali/route-map-http/lib/url-parser.js
// RegExp author cite: http://stackoverflow.com/a/11976301
const DEFAULT_PROTOCOL = 'http:';
const URL_REG = new RegExp(
  '^([a-z0-9-]+\:)?' + //protocol
  '[/]{2}' + //slash x 2
  '(?:([^@/:\?]+)(?::([^@/:]+))?@)?' + //username:password@
  '([^:/?#]+)' + //hostname
  '(?:[:]([0-9]+))?' + //port
  '([/][^?#;]*)?' + //pathname
  '(?:[?]([^#]*))?' + //search
  '([#][^?]*)?$' //hash
  , 'i');

function UrlObject(url) {

  this.parse(url);
}

UrlObject.prototype = {

  parse: function(url) {
    //  1       2       3       4           5       6       7       8               9           10                  11
    // ["ftp", "user", "pass", "www.cs", "server", "com", "8080", "/dir1/dir2/", "file.php", "param1=value1", "hashtag"]
    // ftp://user:pass@www.cs.server.com:8080/dir1/dir2/file.php?param1=value1#hashtag
    var matches = url.match(URL_REG) || [];
    this.protocol = (matches[1] || DEFAULT_PROTOCOL);
    this.username = matches[2];
    this.password = matches[3];
    this.hostname = matches[4] || '';
    this.port = matches[5] || '';
    this.pathname = matches[6] || '/';
    this.search = matches[7] ? '?' + matches[7] : '';
    this.hash = matches[8] || '';
    this.host = this.hostname + (this.port ? ':' + this.port : '');
    this.origin = this.protocol + '//' + this.host;
    this.href = this.toString();
  }

};

module.exports = UrlObject;
