// _universal-env@1.0.7@universal-env/lib/index.js
"use strict";

exports.__esModule = true;
exports.isIOS = exports.isAndroid = exports.isWeb = exports.isWechatApp = exports.isMiniApp = exports.isReactNative = exports.isWeex = exports.isNode = void 0;
// compatibility taobao mp
var isTaoWebview = typeof __sfc__ !== 'undefined';
var isWebPure = typeof navigator === 'object' && (navigator.appCodeName === 'Mozilla' || navigator.product === 'Gecko');
var isNode = typeof process !== 'undefined' && !!(process.versions && process.versions.node);
exports.isNode = isNode;
var isWeex = typeof callNative === 'function' || typeof WXEnvironment === 'object' && WXEnvironment.platform !== 'Web';
exports.isWeex = isWeex;
var isReactNative = typeof __fbBatchedBridgeConfig !== 'undefined';
exports.isReactNative = isReactNative;
var isMiniApp = typeof my === 'object' && typeof my.getSystemInfo !== 'undefined' && !isTaoWebview;
exports.isMiniApp = isMiniApp;
var isWechatApp = typeof wx === 'object' && typeof wx.getSystemInfo !== 'undefined'; // In taobao mp'webview __sfc__ is an object

exports.isWechatApp = isWechatApp;
var isWeb = isWebPure && !isMiniApp && !isWechatApp || isTaoWebview;
exports.isWeb = isWeb;
var systemInfo = {};

if (isMiniApp) {
  systemInfo = my.getSystemInfoSync();
}

if (isWechatApp) {
  systemInfo = wx.getSystemInfoSync();
}

if (isWeex) {
  systemInfo = navigator;
}

var isAndroid = function () {
  if (isMiniApp || isWechatApp || isWeex) {
    var platform = systemInfo.platform || '';
    return platform.toLowerCase() === 'android';
  } else if (isWeb) {
    return Boolean(navigator.userAgent.match(/android/i));
  }

  return false;
}();

exports.isAndroid = isAndroid;

var isIOS = function () {
  if (isMiniApp || isWechatApp || isWeex) {
    return ['ios', 'iOS', 'iPhone OS'].indexOf(systemInfo.platform) > -1;
  } else if (isWeb) {
    return Boolean(navigator.userAgent.match(/(iphone|ipod|ipad)/i));
  }

  return false;
}();

exports.isIOS = isIOS;
